### PR TITLE
socket: implement shutdown(2)

### PR DESCRIPTION
**Pull request by Andy-Python-Programmer**

### CODE REVIEW

File: options/posix/generic/sys-socket-stubs.cpp [hunk 1]
```diff
@@ -187,9 +187,9 @@ int setsockopt(int fd, int layer, int number,
 	return mlibc::sys_setsockopt(fd, layer, number, buffer, size);
 }
 
-int shutdown(int, int) {
-	mlibc::infoLogger() << "mlibc: shutdown() is a no-op!" << frg::endlog;
-	return 0;
+int shutdown(int sockfd, int how) {
+	MLIBC_CHECK_OR_ENOSYS(mlibc::sys_shutdown, -1);
+	return mlibc::sys_shutdown(sockfd, how);
```
Comment (no92) [inline options/posix/generic/sys-socket-stubs.cpp:192]: This is incorrect; to quote the man page,

> On success, zero is returned.  On error, -1 is returned, and errno is set to indicate the error.

You can copy the implementation for this from e.g. starting at line 172.

Comment (no92) [reply options/posix/generic/sys-socket-stubs.cpp:192]: Also, please change this to use the `auto sysdep = MLIBC_CHECK_OR_ENOSYS(...)` idiom, as done in e.g. `net-if-stubs`.

**Review by no92 — CHANGES_REQUESTED** [review]

File: options/posix/generic/sys-socket-stubs.cpp (hunk 1 above)
Comment (Andy-Python-Programmer) [reply options/posix/generic/sys-socket-stubs.cpp]: Fixed.

**Review by ikbenlike — APPROVED** [review]
LGTM

**Review by Dennisbonke — COMMENTED** [review]
LGTM but could you also add the Linux sysdep for it while you're at it? Always nice to improve the Linux port and it should be just a few lines.

**Pull request merged**